framework/test/mocha/integration/blocks/process/on_receive_block.js

```diff
@@ -126,12 +123,10 @@ describe('integration test (blocks) - process receiveBlockFromNetwork()', () =>
 		function getNextForger(offset, seriesCb) {
 			offset = !offset ? 0 : offset;
 			const round = slots.calcRound(last_block.height + 1);
-			library.modules.rounds
-				.generateDelegateList(round, getKeysSortByVote)
-				.then(delegateList => {
-					const nextForger = delegateList[(slot + offset) % ACTIVE_DELEGATES];
-					return seriesCb(nextForger);
-				});
+			library.modules.dpos.getRoundDelegates(round).then(delegateList => {
```

Comment:
if it doesn't require a big refactor, maybe we can await here ?

Reply:
I would suggest creating a new issue to replace use of callbacks in this file and use async/await consistently.